### ISSUE TITLE
[CPTDAWN+CPTGRIM] Forbidden Lands 2+3 - Crashes MBBSEmu when entering module(s)

### ISSUE THREAD

**Issue by BlaZZZed1980** [enusbaum/MBBSEmu#152]
**Module Information**
* CPTDAWN + CPTGRIM
* Forbidden Lands 2+3

**Describe the bug**
MBBSEmu crashes when a user attempts to enter either of these modules, returning identical error messages

**To Reproduce**
1. Load CPTDAWN or CPTGRIM with the latest release of MBBSEmu
2. Connect and attempt to enter either module
3. MBBSEmu will crash

**Expected behavior**
Previous stable release allowed entry to both of these modules with no errors.

**Console Error Message**
```
Unhandled exception. System.Exception: Type Requested not UInt16
   at MBBSEmu.HostProcess.ExecutionUnits.ExecutionUnit.Execute(IntPtr16 entryPoi
nt, UInt16 channelNumber, Boolean simulateCallFar, Boolean bypassState, Queue`1
initialStackValues, UInt16 initialStackPointer)
   at MBBSEmu.Module.MbbsModule.Execute(IntPtr16 entryPoint, UInt16 channelNumbe
r, Boolean simulateCallFar, Boolean bypassSetState, Queue`1 initialStackValues,
UInt16 initialStackPointer)
   at MBBSEmu.HostProcess.MbbsHost.WorkerThread()
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionCo
ntext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionCo
ntext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
```
-----The above capture was for CPTDAWN.  I think the CPTGRIM error is identical, but I've posted it below just in case it isn't...
```
Unhandled exception. System.Exception: Type Requested not UInt16
   at MBBSEmu.HostProcess.ExecutionUnits.ExecutionUnit.Execute(IntPtr16 entryPoi
nt, UInt16 channelNumber, Boolean simulateCallFar, Boolean bypassState, Queue`1
initialStackValues, UInt16 initialStackPointer)
   at MBBSEmu.Module.MbbsModule.Execute(IntPtr16 entryPoint, UInt16 channelNumbe
r, Boolean simulateCallFar, Boolean bypassSetState, Queue`1 initialStackValues,
UInt16 initialStackPointer)
   at MBBSEmu.HostProcess.MbbsHost.WorkerThread()
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionCo
ntext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionCo
ntext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
```
-----The above capture was for CPTGRIM.

**Software Information:**
- OS: Windows 7 32-bit
- MBBSEmu v1.0-alpha-092720 (mbbsemu-win-x32-092620-4.zip)

**Comment by enusbaum** [comment]
This looks like a side effect of the x87 Floating Point fixes that went in #119 

The issue is the instruction FILD is loading a 64-bit Integer into the FPU Stack. Currently the `FILD` instruction only supports 16-bit values, so the FILD and memory operations in the CPU core will need to be enhanced to support 64-bit.

![image](https://user-images.githubusercontent.com/1139047/94727548-e461ef80-032c-11eb-8cc6-b268076e43b3.png)

**Comment by enusbaum** [comment]
Just submitted a PR to fix this and any other Floating Point related crashes that were introduced in the last release 👍
